app/mci/pkg/gcp/loadbalancer/loadbalancersyncer.go

```diff
@@ -98,10 +102,17 @@ func (l *LoadBalancerSyncer) CreateLoadBalancer(ing *v1beta1.Ingress, forceUpdat
 		return multierror.Append(err, igErr)
 	}
 	// Create backend service. This should always be called after the health check since backend service needs to point to the health check.
-	if beErr := l.bss.EnsureBackendService(l.lbName, ports, healthChecks, namedPorts, igs); beErr != nil {
-		return multierror.Append(err, beErr)
+	backendServices, beErr := l.bss.EnsureBackendService(l.lbName, ports, healthChecks, namedPorts, igs)
+	if beErr != nil {
+		// Aggregate errors and return all at the end.
+		err = multierror.Append(err, beErr)
 	}
-	return nil
+	umErr := l.ums.EnsureURLMap(l.lbName, ing, backendServices)
+	if umErr != nil {
+		// Aggregate errors and return all at the end.
+		err = multierror.Append(err, umErr)
+	}
+	return err
 }
```

Comment:
How/where do we delete the healthcheck if, for example, we can successfully create the HC but the backendservice fails.

Reply:
Good point. We dont. User can retry with `mci create` again or delete resources by running `mci delete`.

Reply:
I'm not sure that this is a good user experience. Don't want to block the PR but let's file an issue to get back to this soon.

Reply:
Thx, filed https://github.com/GoogleCloudPlatform/k8s-multicluster-ingress/issues/8